File: apps/web/src/avatars.ts
// Mirrors services/api/app/avatars.py's AVATAR_KEYS - kept in sync by
// hand (see that file's own comment on why: no shared-types package
// crosses the Python/TypeScript boundary in this monorepo). The backend
// only ever sees/validates the key; this file owns the actual glyph.
export const AVATAR_OPTIONS: { key: string; emoji: string; label: string }[] = [
  { key: "astronaut", emoji: "🧑‍🚀", label: "Astronaut" },
  { key: "hero", emoji: "🦸", label: "Hero" },
  { key: "wizard", emoji: "🧙", label: "Wizard" },
  { key: "robot", emoji: "🤖", label: "Robot" },
  { key: "cat", emoji: "🐱", label: "Cat" },
  { key: "fox", emoji: "🦊", label: "Fox" },
  { key: "dragon", emoji: "🐉", label: "Dragon" },
  { key: "alien", emoji: "👽", label: "Alien" },
  { key: "ninja", emoji: "🥷", label: "Ninja" },
  { key: "vampire", emoji: "🧛", label: "Vampire" },
  { key: "lion", emoji: "🦁", label: "Lion" },
  { key: "panda", emoji: "🐼", label: "Panda" },
];

const BY_KEY = new Map(AVATAR_OPTIONS.map((option) => [option.key, option]));

export function avatarEmoji(key: string | null | undefined): string {
  return (key && BY_KEY.get(key)?.emoji) || "🧑‍🚀";
}

File: services/api/app/avatars.py
# The set of characters a player can pick as their avatar. Stored as
# keys (not emoji) on the User row - the frontend owns the actual glyph
# for each key (apps/web/src/avatars.ts), kept in sync by hand since
# there's no shared-types package crossing the Python/TypeScript
# boundary here (see ENGINEERING.md's "modular monorepo, not microservices" -
# this is the one place that boundary costs a little duplication).
AVATAR_KEYS = [
    "astronaut",
    "hero",
    "wizard",
    "robot",
    "cat",
    "fox",
    "dragon",
    "alien",
    "ninja",
    "vampire",
    "lion",
    "panda",
]

DEFAULT_AVATAR = "astronaut"
